weave/trace_server_bindings/remote_http_trace_server.py

```diff
@@ -161,6 +161,41 @@ def _flush_calls(
             self._flush_calls(batch[split_idx:], _should_update_batch_size=False)
             return
 
+        final_batch = []
+        if encoded_bytes > CLICKHOUSE_SINGLE_ROW_INSERT_BYTES_LIMIT:
+            # ensure that there isn't a single row that is too big
+            seen_bytes = 0
+            for i, item in enumerate(batch):
+                # below takes tens of miliseconds for the large payloads we skip
+                bytes_size = len(item.model_dump_json().encode("utf-8"))
+                seen_bytes += bytes_size
+                if bytes_size > CLICKHOUSE_SINGLE_ROW_INSERT_BYTES_LIMIT:
+                    mb = bytes_size / (1024 * 1024)
+                    id_, mode = (
+                        (item.req.start.id, "start")
+                        if isinstance(item, StartBatchItem)
+                        else (item.req.end.id, "end")
+                    )
+                    print(
+                        f"Dropping call {mode} ({id_}) with size {mb:.2f} MiB larger than "
+                        f"{CLICKHOUSE_SINGLE_ROW_INSERT_BYTES_LIMIT / (1024 * 1024):.2f} MiB."
+                        " If logging images, please encode them as `PIL.Image`"
```

Comment:
It seems like we want to re-use this error when encountering a failed batch of that specific error as well.